test/libsolidity/semanticTests/externalSource/source_name_starting_with_dots.sol

```diff
@@ -0,0 +1,22 @@
+==== ExternalSource: /Users/alex/git/solidity/cmake-build-debug/test/tools/a.sol=_source_name_starting_with_dots/dot_a.sol ====
+==== ExternalSource: /Users/alex/git/solidity/cmake-build-debug/test/b.sol=_source_name_starting_with_dots/dot_dot_b.sol ====
+==== ExternalSource: _source_name_starting_with_dots/dir/a.sol ====
+==== ExternalSource: _source_name_starting_with_dots/b.sol ====
+==== ExternalSource: _source_name_starting_with_dots/dir/contract.sol ====
+import "_source_name_starting_with_dots/dir/contract.sol";
```

Comment:
What about remappings like `ExternalSource: ./a.sol=_source_name_starting_with_dots/dot_a.sol`? I think we should have a test checking how they work. Such a remapping **should not** match `import "./a.sol"`, That's because the `./` part gets replaced by the path to the importing module when `CompilerStack` evaluates the import.

Reply:
Exactly. It looks like that `import "./a.sol"` would become `import "a.sol"`, that is not found. But within `externalSource/_source_name_starting_with_dots/dir/contract.sol`, we do `import "./a.sol";` and `import "../b.sol";`, that becomes `import "dir/a.sol";` and `import "b.sol";` So I think what you described is already tested, not sure what you mean. E.g. if `import "./a.sol";` would import the defined `_source_name_starting_with_dots/dot_a.sol`, `contract A` would not be found.

Reply:
Yeah, it's being tested because the `./a.sol=` remapping is present now. When I commented it was not there. I think it was because of https://github.com/ethereum/solidity/pull/11110#discussion_r618667060.